### ISSUE TITLE
CKV_GCP_73 reports a false-positive check when a dynamic rule is used

### ISSUE THREAD

**Issue by calexandre** [bridgecrewio/checkov#4740]
**Describe the issue**

- CKV_GCP_73 reports a false-positive check when a dynamic rule is used
- In the example below we have the rule as explained in the documentation, followed by a dynamic sequence.
- As a temporary fix we were forced to add a skip comment

**Examples**
```hcl
resource "google_compute_security_policy" "default" {
  name = "${var.service_name}-cloudrun-security-policy"

  rule {
    action      = "deny(403)"
    priority    = "1"
    description = "cve-canary"
    match {
      expr {
        expression = "evaluatePreconfiguredExpr('cve-canary')"
      }
    }
  }

  dynamic "rule" {
    for_each = var.cloud_armor_rules
    content {
      action   = rule.value.action
      priority = rule.value.priority
      match {
        versioned_expr = rule.value.match.versioned_expr
        dynamic "config" {
          for_each = rule.value.match.config != null ? [1] : []
          content {
            src_ip_ranges = rule.value.match.config.src_ip_ranges
          }
        }

        dynamic "expr" {
          for_each = rule.value.match.expr != null ? [1] : []
          content {
            expression = rule.value.match.expr.expression
          }
        }
      }
      description = rule.value.description
    }

  }

  rule {
    action      = "deny(403)"
    priority    = "2147483647"
    description = "Default rule: deny all"

    match {
      versioned_expr = "SRC_IPS_V1"
      config {
        src_ip_ranges = ["*"]
      }
    }
  }
}
```

**Version (please complete the following information):**
 - `2.3.75`
